### PR TITLE
Merge 2.5.4 fixes

### PR DESCRIPTION
**Pull request by cloudw**

### CODE REVIEW

File: metaflow/current.py [hunk 1]
```diff
@@ -85,7 +65,7 @@ def is_running_flow(self):
         return self._is_running
 
     @property
-    def flow(self) -> FlowSpec:
+    def flow(self):
```
Comment (talebzeghmi) [inline metaflow/current.py:68]: style: return type ` -> FlowSpec:`

File: metaflow/util.py [hunk 2]
```diff
@@ -167,14 +167,7 @@ def get_username():
     could not be determined.
     """
     # note: the order of the list matters
-<<<<<<< HEAD
-    if METAFLOW_USER:
-        return METAFLOW_USER
```
Comment (talebzeghmi) [inline metaflow/util.py:172]: You missed this! `METAFLOW_USER` added in https://github.com/zillow/metaflow/pull/12

**Review by talebzeghmi — CHANGES_REQUESTED** [review]

File: metaflow/plugins/kfp/kfp_decorator.py [hunk 3]
```diff
@@ -76,20 +69,27 @@ def myStep(self):
...
+    def step_init(self, flow, graph, step, decos, environment, flow_datastore, logger):
         if self.attributes["preceding_component"] is not None:
             node = graph[step]
             if step == "start":
                 raise KfpException(
                     "A @kfp preceding_component cannot be on the start step."
                 )
 
-            if len(node.in_funcs) > 1 or graph[node.in_funcs[0]].type != "linear":
+            linear_types = (  # Excluding "join" type to avoid complexity of merge_artifacts
```
[the remark this answers is not in the document]
Comment (cloudw) [reply metaflow/plugins/kfp/kfp_decorator.py:80]: Nothing is changed as the previous "linear" type is now "start" and "linear" type. I was trying to add my understanding while reading the code. 

For join type node (C), say we have two node going into this node A->C and B->C, we usually combine self.var's in C using [merge_artifact](https://github.com/Netflix/metaflow/blob/master/metaflow/flowspec.py#L350). Suppose we have 
```
A.x = 1
A.y = 2
B.z = 3
```
Ideally we want to make sure `x`, `y` and `z` are all available in `preceding_component`. However the `preceding_component` happens before C and `merge_artifact` might not have happened yet. We don't have a good previous step to get artifacts/variables from.

File: metaflow/util.py (hunk 2 above)
Comment (cloudw) [reply metaflow/util.py:172]: Ah. I was thinking why this was needed and myself in https://github.com/zillow/metaflow/pull/12/files#r490528824 answered it. Fixing...

Comment (cloudw) [reply metaflow/util.py:172]: Looks like something that can be contributed back to `netflix/metaflow`

**Comment by cloudw** [comment]
Test will fail in this branch since black upgrade is in https://github.com/zillow/metaflow/pull/164 to reduce number of files to be reviewed.

**Review by talebzeghmi — APPROVED** [review]
nice work!

**Pull request merged**